src/systolic_array/commit.cpp

```diff
@@ -34,7 +36,7 @@ void Commit::setParams() {
       { 0, 0, 0, 0 },
       { 1, accel.outputRows, accel.outputCols, accel.peArrayCols });
   // Move the iterator to the correct starting place.
-  iter += { 0, 0, 0, accel.lineSize / accel.elemSize * id };
+  iter += { 0, 0, id, 0 };
```

Comment:
Can you elaborate why this is the correct place now? I'm a bit confused by the previous code - were we relying on the iterator math to carry over the addition from dim[0] to dim[1]?

Reply:
I was confused by the previous code too.. Since one PE row is responsible for producing an output channel, commit unit i should start at [0, 0, i, 0] and once it finishes that channel, it advances to [0, 0, 2*i, 0].

Reply:
Forgot to mention here the layout format used in the systolic array is NHWC.